script.js

```diff
@@ -0,0 +1,378 @@
+  // go through the temphand array
+  for (let j = 0; j < player.tempHand.length; j += 1) {
+    // if the card at temphand[j] matches the clicked card, then remove temphand[j] from the tempHand array
+    if ((clickedCard.display === player.tempHand[j].display) && (clickedCard.suit === player.tempHand[j].suit)) {
+      player.tempHand.splice(j, 1);
+      // console.log('splicing this card');
+      // console.log(player.tempHand[j]);
+      // const keepDiv = document.getElementById('keepCardIndicationID');
+      const keepDiv = cardElement.querySelector('div');
+      console.log(keepDiv);
+      cardElement.removeChild(keepDiv);
+      return;
+    }
+  }
+  // if user clicks a card, add it to new array
+  // console.log('pushing....');
+  player.tempHand.push(clickedCard);
+  cardElement.appendChild(keepCardIndication);
+};
+
+// ----------determine bet type------------------
+const getStakes = () => {
+  const stakesArray = [{}, {}, {}, {}];
+  // make first object in array
+  stakesArray[0] = {
+    straightFlush: 50,
+    fourOfAKind: 25,
+    fullHouse: 9,
+    flush: 6,
+    straight: 4,
+    threeOfAKind: 3,
+    twoPair: 2,
+    jacksOrLarger: 1,
+    lose: -1,
+  };
+  // using first object, makes subsequent objects by using multiples of first object's properties.
+  for (let i = 1; i < stakesArray.length; i += 1) {
+    stakesArray[i] = {
+      straightFlush: stakesArray[0].straightFlush * (i + 1),
+      fourOfAKind: stakesArray[0].fourOfAKind * (i + 1),
+      fullHouse: stakesArray[0].fullHouse * (i + 1),
+      flush: stakesArray[0].flush * (i + 1),
+      straight: stakesArray[0].straight * (i + 1),
+      threeOfAKind: stakesArray[0].threeOfAKind * (i + 1),
+      twoPair: stakesArray[0].twoPair * (i + 1),
+      jacksOrLarger: stakesArray[0].jacksOrLarger * (i + 1),
+      lose: stakesArray[0].jacksOrLarger * (i + 1),
+    };
+  }
+  console.log(stakesArray);
+  return stakesArray[bidTracker];
+};
+
+const clickRestart = () => {
+  console.log(`bidtracker is ...${bidTracker}`);
+  // reset bidTracker to 0
+  bidTracker = 0;
+  // display new bidtracker value
+  document.getElementById('bidDisplayID').innerText = bidTracker;
+
+  console.log(`bidtracker is ...${bidTracker}`);
+  // reset bidding button to ensure it is click-able
+  document.getElementById('bidButton').removeAttribute('disabled');
+  // initialise the game
+  initGame();
+};
+//= ===========GAME FLOW========================
+const initGame = () => {
+  // build the board
+  buildBoardElements();
+
+  // initialise the player's hand+ create his cards' elements
+  intialiseHand();
+  createCardsElements();
+
+  // Display the user's credits
+
+  // Logic to decide whether player is going forward with his current hand,
+  // or if he wants to return/draw new cards:
+
+  // if player clicks the draw button and has selected cards to keep, then draw new cards for him;
+  document.getElementById('dealOrDrawButtonID').addEventListener('click', () => {
```

Comment:
Consider decomposing game logic into helper function to simplify the higher-level game logic. In general, separating the UI (DOM) and app (gameplay) logic will make code more readable and maintainable.